{
  return function.apply(e instanceof ExecutionException ? (Exception)e.getCause() : e);
}
